hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -707,7 +707,8 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
           clients.run(client -> client.getDatabase(tableIdentifier.namespace().levels()[0]));
       if (databaseData.getLocationUri() != null) {
         // If the database location is set use it as a base.
-        return String.format("%s/%s", databaseData.getLocationUri(), tableIdentifier.name());
+        String stripLocationUri = LocationUtil.stripTrailingSlash(databaseData.getLocationUri());
+        return String.format("%s/%s", stripLocationUri, tableIdentifier.name());
```

Comment:
@prakhar10 can you add a corresponding unit test to illustrate the failure before and succeeded after the change?

Reply:
maybe add something like [this](https://github.com/apache/iceberg/blob/main/aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java#L134-L142) to https://github.com/apache/iceberg/blob/c8c61a3f6b60499f09e5047c5c9d8a3c02d9fb72/hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java#L95